src/lib/containers/useImmutableTableQuery.ts

```diff
@@ -0,0 +1,171 @@
+import { useCallback, useEffect, useState } from 'react'
+import { QueryBundleRequest } from '../utils/synapseTypes/Table/QueryBundleRequest'
+import { cloneDeep } from 'lodash-es'
+import * as DeepLinkingUtils from '../utils/functions/deepLinkingUtils'
+import { DEFAULT_PAGE_SIZE } from '../utils/SynapseConstants'
+import { parseEntityIdAndVersionFromSqlStatement } from '../utils/functions/sqlFunctions'
+
+export type ImmutableTableQueryResult = {
+  /** The ID of the table parsed from the SQL query */
+  entityId: string
+  /** The version number of the table parsed from the SQL query */
+  versionNumber?: number
+  getInitQueryRequest: () => QueryBundleRequest
```

Comment:
Ideally, this would be an immutable object, rather than a function that returns a clone. But that's a larger change where we may have to commit to a new dependency (until we get [Records and Tuples](https://github.com/tc39/proposal-record-tuple) 🙂)